### PR TITLE
Fix S113 Probable use of requests call without timeout

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

**Comment by brianhelba** [comment]
This may not capture all use of `requests`, but it's at least an improvement.

File: dandiapi/api/doi.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def create_doi(version: Version) -> str:
                     settings.DANDI_DOI_API_USER,
                     settings.DANDI_DOI_API_PASSWORD,
                 ),
+                timeout=5,
```
Comment (yarikoptic) [inline dandiapi/api/doi.py:47]: do we have any timing measurements so far for this endpoint?
5 seconds sounds like a lot, but not really to me, and I would hate for us to fail to mint a DOI due to some "slow processing" by datacite. I would have made it 60 or so or do you see this process blocking some other legitimate load?

Comment (yarikoptic) [reply dandiapi/api/doi.py:47]: What about

```suggestion
                timeout=30,
```

**Comment by yarikoptic** [comment]
FTR similar pr in client: https://github.com/dandi/dandi-cli/pull/1369

File: dandiapi/api/doi.py (hunk 1 above)
Comment (brianhelba) [reply dandiapi/api/doi.py:47]: Good suggestion, thanks. I'm updating this code and the scripts to timeout in 30 seconds to be conservative, but I'll keep the tests at 5 seconds, so they fail faster.

**Review by mvandenburgh — APPROVED** [review]

**Pull request merged**